lib/widgets/lightbox.dart

```diff
@@ -197,22 +199,207 @@ class _LightboxPageState extends State<_LightboxPage> {
+          // take user back to content message list.
+          barrierDismissible: false,
+          onContinue: () {
+            Navigator.pop(context); // Pops the dialog
+            Navigator.pop(context); // Pops the lightbox
+          });
+      }
+    }
+  }
+
+  @override
+  void dispose() {
+    _deinitialize();
+    super.dispose();
+  }
+
+  void _deinitialize() {
+    _controller?.removeListener(_handleVideoControllerUpdates);
+    _controller?.dispose();
+    _controller = null;
+  }
+
+  void _handleVideoControllerUpdates() {
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _LightboxPageLayout(
+      routeEntranceAnimation: widget.routeEntranceAnimation,
+      message: widget.message,
+      buildBottomAppBar: (context, color, elevation) =>
+        _controller == null
+          ? null
+          : BottomAppBar(
+              height: 150,
+              color: color,
+              elevation: elevation,
+              child: Column(mainAxisAlignment: MainAxisAlignment.end, children: [
+              Row(children: [
+                Text(_formatDuration(_controller!.value.position),
+                  style: const TextStyle(color: Colors.white)),
+                Expanded(child: Slider(
+                  value: _controller!.value.position.inSeconds.toDouble(),
+                  max: _controller!.value.duration.inSeconds.toDouble(),
+                  activeColor: Colors.white,
+                  onChanged: (value) {
+                    _controller!.seekTo(Duration(seconds: value.toInt()));
+                  })),
+                Text(_formatDuration(_controller!.value.duration),
+                  style: const TextStyle(color: Colors.white)),
+                ]),
+              IconButton(
+                onPressed: () {
+                  if (_controller!.value.isPlaying) {
+                    _controller!.pause();
+                  } else {
+                    _controller!.play();
+                  }
+                },
+                icon: Icon(
+                  _controller!.value.isPlaying
+                    ? Icons.pause_circle_rounded
+                    : Icons.play_circle_rounded,
+                  size: 50)),
+              ])),
+      child: SafeArea(
+        child: Center(
+          child: Stack(alignment: Alignment.center, children: [
+            if (_controller != null && _controller!.value.isInitialized)
+              AspectRatio(
+                aspectRatio: _controller!.value.aspectRatio,
+                child: VideoPlayer(_controller!)),
+            if (_controller == null || !_controller!.value.isInitialized || _controller!.value.isBuffering)
+              const SizedBox(
+                width: 32,
+                height: 32,
+                child: CircularProgressIndicator(color: Colors.white)),
+            ]))));
+  }
+
+  String _formatDuration(Duration value) {
```

Comment:
nit: other methods come before `build`:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#order-other-class-members-in-a-way-that-makes-sense

Separately: this is naturally `static` — it doesn't care about any data from the class instance — so let's declare it that way.